src/main/java/com/mathworks/ci/actions/RunMatlabAction.java

```diff
@@ -0,0 +1,81 @@
+package com.mathworks.ci.actions;
+
+/**
+ * Copyright 2024, The MathWorks Inc.
+ *
+ */
+
+import com.mathworks.ci.BuildArtifactAction;
+import com.mathworks.ci.BuildConsoleAnnotator;
+import com.mathworks.ci.MatlabBuilderConstants;
+import com.mathworks.ci.utilities.MatlabCommandRunner;
+import hudson.FilePath;
+import hudson.model.Run;
+import org.apache.commons.lang.RandomStringUtils;
+
+import java.io.File;
+import java.io.IOException;
+
+public class RunMatlabAction {

```

Comment:
If this is going to be the parent class for the other three actions would you be open to naming it `MatlabAction` instead of `RunMatlabAction`? I think it's just a little bit cleaner.

Reply:
I guess we could simply `MatlabAction` as well